toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/attachment/AttachmentElementState.kt

```diff
@@ -182,19 +203,27 @@ internal class FormAttachmentState(
      */
     val type: AttachmentType = getAttachmentType(name)
 
-    constructor(attachment: FormAttachment, scope: CoroutineScope) : this(
+    constructor(
+        element: AttachmentElementState,
+        attachment: FormAttachment,
+        scope: CoroutineScope
+    ) : this(
         name = attachment.name,
         size = attachment.size,
+        elementStateId = element.id,
         loadStatus = attachment.loadStatus,
         onLoadAttachment = attachment::load,
```

Comment:
If a Loadable is FailedToLoad state, calling `load()` on it will immediately fail. You must call retryLoad() to load a previously failed Loadable.

```suggestion
        onLoadAttachment = attachment::retryLoad,
```

Reply:
thanks, updated!